src/main/java/jenkins/advancedqueue/jobinclusion/strategy/FolderBasedJobInclusionStrategy.java

```diff
@@ -62,19 +66,70 @@ public ListBoxModel getListFolderItems() {
 
 	};
 
+	static public class JobPattern {
```

Comment:
I've tried to make it work with a normal `String` instead of the `JobPattern` but to no avail, it seems like I'm missing something obvious here, probably due to my low understanding of the ecosystem. I had to go with the `@Restricted` route. I'm not sure what constructor modifications you mean, though.